READER - Card Refresh

davemart-in opened the pull request:
## Proposed Changes

The following PR comprises a handful of related issues, including:

- [x] #66626
- [x] #66623
- [x] #66624
- [x] #66625
- [x] #66622
- [x] #66618
- [x] #67166
- [x] #66619
- [x] #66621
- [x] #67564
- [x] #66620
- [x] https://github.com/Automattic/wp-calypso/pull/68209


The goal being to update the visual design of all reader cards.

### Before

Here's an example of what the reader card looked like before these changes:

![CleanShot 2022-08-17 at 10 39 07](https://user-images.githubusercontent.com/5634774/185165741-345acded-1437-4cf9-8ff0-03aed566677b.png)

### After

Here's what they will look like after:

![CleanShot 2022-08-17 at 10 40 35](https://user-images.githubusercontent.com/5634774/185166595-d924c731-4763-4702-978a-d0b237964b98.png)

davemart-in committed:
Reduce width of reader container to be 600px Addressing issue in #66626

davemart-in committed:
Reduce width of reader container to be 600px Addressing issue in #66626

davemart-in committed:
Add domain next to date - Issue outlined in #66623

davemart-in committed:
Add domain next to date - Issue outlined in #66623

roo2 committed:
WIP move ellipsis menu to the top of reader card (#66749)

roo2 committed:
WIP move ellipsis menu to the top of reader card (#66749)

roo2 commented:
We'll have to look at the way that very long tags wrap. It's currently looking quite cramped with the reduced width and the hamburger menu on top cc @ollierozdarz @davemart-in 
<img width="619" alt="Screen Shot 2022-08-24 at 11 56 33 am" src="https://user-images.githubusercontent.com/22446385/186300799-4cc99b49-98d0-40cf-be63-30628baab220.png">

davemart-in commented:
@roo2 we've got an issue to move them to their own line and apply a bit of styling: #66622

Should be an easy one.

eoigal committed:
Remove combine card functionality from reader stream

eoigal committed:
Remove combine card functionality from reader stream

eoigal committed:
Revert "Remove combine card functionality from reader stream"

This reverts commit 3915c9b40f12eb40c1f9bf412cd8d8be6aaf6205.

eoigal committed:
Revert "Remove combine card functionality from reader stream"

This reverts commit 3915c9b40f12eb40c1f9bf412cd8d8be6aaf6205.

eoigal committed:
Remove combine card functionality from reader stream

eoigal committed:
Remove combine card functionality from reader stream

ollierozdarz commented:
@roo2 this is how we've defined tags to be styled going forward:

![image](https://user-images.githubusercontent.com/67085494/186575914-9f324877-cba9-44b9-a465-ed68eb22b9be.png)

davemart-in committed:
Update separator styles on responsive mobile

Resolves changes in #66618

davemart-in committed:
Update separator styles on responsive mobile

Resolves changes in #66618

eoigal committed:
Add/reader tags newline 3 (#67117)

* Update reader tags

* Updated width of tags container

With recent updates to the more menu, a few minor adjustments needed to be made.

Co-authored-by: Dave Martin <davemart-in@users.noreply.github.com>

eoigal committed:
Add/reader tags newline 3 (#67117)

* Update reader tags

* Updated width of tags container

With recent updates to the more menu, a few minor adjustments needed to be made.

Co-authored-by: Dave Martin <davemart-in@users.noreply.github.com>

jordesign committed:
Set feed header elements in correct source order and general layout

jordesign committed:
Set feed header elements in correct source order and general layout

jordesign committed:
Align text and set profile image size

jordesign committed:
Align text and set profile image size

jordesign committed:
set max width on main column

jordesign committed:
Center header content and set bottom margin

jordesign committed:
Center header content and set bottom margin

jordesign committed:
Fix up vertical padding between header elements

jordesign committed:
Fix up vertical padding between header elements

jordesign committed:
Set larger font size for Site Title

jordesign committed:
Set larger font size for Site Title

jordesign committed:
Set font size for site description

jordesign committed:
Set font size for site description

jordesign committed:
Change position of 'Back' button in Reader

jordesign committed:
Change position of 'Back' button in Reader

davemart-in committed:
Added focus styles for settings link. Also adjusted line height of two buttons

davemart-in committed:
Added focus styles for settings link. Also adjusted line height of two buttons

davemart-in committed:
Update line clamping on excerpts

davemart-in committed:
Update line clamping on excerpts

davemart-in committed:
Merge branch 'try_new_reader_site_layout' into update/reader-card-refresh

davemart-in committed:
Remove search bar and search suggestions from following page

davemart-in committed:
Remove search bar and search suggestions from following page

eoigal committed:
Reader Card Refresh: Stack media vertically and make full width (#67387)

* Add featured images component to build flex containers for featured images

* Update gallery to work as a slideshow

* fix missing height

* Add state to handle changing gallery images

* Sort images by width size and update number of images supported by gallery

* fix bug with video thumbnails

* fix related cards width bug

* fix video embeds thumbnails

* add lint changes

* fix lint issue

* add filtering by image ratio

* refactor getting suitable images into utility function - filter duplicates

* Remove use of isDiscover

* fix gallery click behaviour

* tidy css

* fix missing post prop

* refactor gallery to only re-render images when clicked on

* Enable photo posts again but increase height to match featured image

* Added animation transition to gallery

* Fix gallery issue where images disappear on mobile. Addresses issue mentioned here: https://github.com/Automattic/wp-calypso/pull/67387#issuecomment-1243658345

* Fixed issue with malformed images in conversations. Addresses https://github.com/Automattic/wp-calypso/pull/67387#issuecomment-1243659873

Co-authored-by: Dave Martin <davemart-in@users.noreply.github.com>

eoigal committed:
Reader Card Refresh: Stack media vertically and make full width (#67387)

* Add featured images component to build flex containers for featured images

* Update gallery to work as a slideshow

* fix missing height

* Add state to handle changing gallery images

* Sort images by width size and update number of images supported by gallery

* fix bug with video thumbnails

* fix related cards width bug

* fix video embeds thumbnails

* add lint changes

* fix lint issue

* add filtering by image ratio

* refactor getting suitable images into utility function - filter duplicates

* Remove use of isDiscover

* fix gallery click behaviour

* tidy css

* fix missing post prop

* refactor gallery to only re-render images when clicked on

* Enable photo posts again but increase height to match featured image

* Added animation transition to gallery

* Fix gallery issue where images disappear on mobile. Addresses issue mentioned here: https://github.com/Automattic/wp-calypso/pull/67387#issuecomment-1243658345

* Fixed issue with malformed images in conversations. Addresses https://github.com/Automattic/wp-calypso/pull/67387#issuecomment-1243659873

Co-authored-by: Dave Martin <davemart-in@users.noreply.github.com>

davemart-in committed:
Merge branch 'update/reader-following-page' into update/reader-card-refresh

davemart-in committed:
Added border-radius to feature images

davemart-in committed:
Added border-radius to feature images

davemart-in committed:
Added gutter to left of .reader-post-card__post

davemart-in committed:
Added gutter to left of .reader-post-card__post

davemart-in committed:
Refactored feature image to show entire height of image

davemart-in committed:
Refactored feature image to show entire height of image

davemart-in committed:
Added border-radius to gallery images

davemart-in committed:
Added border-radius to gallery images

davemart-in committed:
Updated avatar size

davemart-in committed:
Updated avatar size

davemart-in committed:
Tightened up padding and margins within .reader-post-card__byline

davemart-in committed:
Tightened up padding and margins within .reader-post-card__byline

davemart-in committed:
Tightened up padding and margins within .reader-post-card__post-details

davemart-in committed:
Tightened up padding and margins within .reader-post-card__post-details

davemart-in committed:
Removed border around feature images

davemart-in committed:
Removed border around feature images

davemart-in committed:
Adjusted follow button positioning

davemart-in committed:
Adjusted follow button positioning

davemart-in committed:
Slightly better styles for 2, 3, and 4 image layouts

davemart-in committed:
Slightly better styles for 2, 3, and 4 image layouts

davemart-in committed:
More refinements to 2, 3, and 4 image cards

davemart-in committed:
More refinements to 2, 3, and 4 image cards

davemart-in committed:
Fixed margin between images on conversations screen

davemart-in committed:
Fixed margin between images on conversations screen

davemart-in committed:
Fixed formatting for 3 image card

davemart-in committed:
Fixed formatting for 3 image card

davemart-in committed:
Added max-height to all single images

davemart-in committed:
Added max-height to all single images

eoigal committed:
Add new Reader engagement Icons (#68103)

* Update reader to use new icons

* Add reader icon

* pass default if no iconsize defined

* Minor CSS tweaks to icons

* Updates to header icon CSS

* Updates styles for open in new tab icon

* remove fill on like star

* add hover to svgs

Co-authored-by: Dave Martin <davemart-in@users.noreply.github.com>

eoigal committed:
Add new Reader engagement Icons (#68103)

* Update reader to use new icons

* Add reader icon

* pass default if no iconsize defined

* Minor CSS tweaks to icons

* Updates to header icon CSS

* Updates styles for open in new tab icon

* remove fill on like star

* add hover to svgs

Co-authored-by: Dave Martin <davemart-in@users.noreply.github.com>

eoigal committed:
Fix excerpt text in reader (#68167)

* Fix Reader excerpts

eoigal committed:
Fix excerpt text in reader (#68167)

* Fix Reader excerpts

eoigal committed:
Fix reader style issues (#68209)

* Fix ellipse menu issues

* Fix padding for x-posts

* Updated a number of color issues Ollie pointed out

* Updated gallery styles

* improve post action highlight

* fix follow conversation hover color

* fix external icon size and position

* fix external icon in post actions

* fix video featured image width

* Fixed font styles and alignment of action buttons

* add follow icon to post actions

* Remove unwanted rounded corner on 3 image card

* Updates to followed sites header

* fixed alignment of i icon

* couple of minor updates

* update follow/following icons and add commented icon

* need to use camelcase

* Fix margin next to follow button

* Update icons to latest designs - add follow/following conversation icons

* Allow follow/following conversation icon to be overridden

* update new icon ref

* Update layout of ellipse menu and add seperators to match designs

* Update icon size and icon class names

* Add invert for new follow icons

* Update follow icons to remove ref to gridicons

* Ref new follow feed icons

* fix bug with passing prop to a tag

* Add copy link action to share popup

* Switch to new Twitter and Facebook icons in share popup

* Updated height of two and three image cards

* Adjusted action line on single post screen

* Fixed conversation page styles

* Two minor accessibility updates

* Fix line clamp issue in FireFox

* fix facebook icon - active state on post actions

* Fine tuning action bar styles

* Make sure mobile web shows images full-width

* Fix 4 image styles on mobile web

* Updated related posts styles

* Fixed like button on single post screen

* Fix four image issue

* Fix following icon - redundant paths in svg

* Fix header cake spacing

* Fix follow icons in comment list

* Fix alignment on post action items on full post page

* no image border-radius

* Cleaned some styles up on single post page

* Updates to action bar on single post page

* Updates to action bar

* Fixes for full post screen on mobile web

* Fixed formatting for two image issue

* Updated search box styles

* Fix missing feed and site icons in reader

* Set feed site_icon

* Add feed icon to full post

* fix bug in feed id

* Add new follow/following feed icons to FollowIcon component

* Avoid using buttonw-com.png as site icon

* use the old like icon on comments and the new like icon for posts

* use primary color for star icon

* fix issue where expanding tags caused hamburger menu to jump out of the flow

* fix the star icon v-alignment on full post

* fix hamburger menu highlighting issue

* Fix like button in comments and tidy up hover state in full post css

* fix typo

Co-authored-by: Dave Martin <davemart-in@users.noreply.github.com>
Co-authored-by: Eru Penkman <roo2@protonmail.com>

eoigal committed:
Fix reader style issues (#68209)

* Fix ellipse menu issues

* Fix padding for x-posts

* Updated a number of color issues Ollie pointed out

* Updated gallery styles

* improve post action highlight

* fix follow conversation hover color

* fix external icon size and position

* fix external icon in post actions

* fix video featured image width

* Fixed font styles and alignment of action buttons

* add follow icon to post actions

* Remove unwanted rounded corner on 3 image card

* Updates to followed sites header

* fixed alignment of i icon

* couple of minor updates

* update follow/following icons and add commented icon

* need to use camelcase

* Fix margin next to follow button

* Update icons to latest designs - add follow/following conversation icons

* Allow follow/following conversation icon to be overridden

* update new icon ref

* Update layout of ellipse menu and add seperators to match designs

* Update icon size and icon class names

* Add invert for new follow icons

* Update follow icons to remove ref to gridicons

* Ref new follow feed icons

* fix bug with passing prop to a tag

* Add copy link action to share popup

* Switch to new Twitter and Facebook icons in share popup

* Updated height of two and three image cards

* Adjusted action line on single post screen

* Fixed conversation page styles

* Two minor accessibility updates

* Fix line clamp issue in FireFox

* fix facebook icon - active state on post actions

* Fine tuning action bar styles

* Make sure mobile web shows images full-width

* Fix 4 image styles on mobile web

* Updated related posts styles

* Fixed like button on single post screen

* Fix four image issue

* Fix following icon - redundant paths in svg

* Fix header cake spacing

* Fix follow icons in comment list

* Fix alignment on post action items on full post page

* no image border-radius

* Cleaned some styles up on single post page

* Updates to action bar on single post page

* Updates to action bar

* Fixes for full post screen on mobile web

* Fixed formatting for two image issue

* Updated search box styles

* Fix missing feed and site icons in reader

* Set feed site_icon

* Add feed icon to full post

* fix bug in feed id

* Add new follow/following feed icons to FollowIcon component

* Avoid using buttonw-com.png as site icon

* use the old like icon on comments and the new like icon for posts

* use primary color for star icon

* fix issue where expanding tags caused hamburger menu to jump out of the flow

* fix the star icon v-alignment on full post

* fix hamburger menu highlighting issue

* Fix like button in comments and tidy up hover state in full post css

* fix typo

Co-authored-by: Dave Martin <davemart-in@users.noreply.github.com>
Co-authored-by: Eru Penkman <roo2@protonmail.com>

eoigal committed:
Merge branch 'trunk' into update/reader-card-refresh

danielbachhuber committed:
Sign Up: Use 'Back to My Home' when user only has 1 site (#68545)

danielbachhuber committed:
Sites Management Page: Fix site list placeholder on mobile (#68585)

mpkelly committed:
Use correct React prop `className` to clear console warning (#68555)

WBerredo committed:
Rename the Themes search empty results event (#68535)

* Update the Themes search empty results event

* Update the event calling from the component to the request to guarantee it will be called only once

eoigal committed:
fix .jsx lint issues

eoigal committed:
fix .jsx lint issues

eoigal committed:
fix flag button style

eoigal committed:
fix flag button style

eoigal committed:
fix lint on img tag

eoigal committed:
fix lint on img tag

eoigal committed:
More lint fixes

eoigal committed:
More lint fixes

eoigal committed:
fix space lint error

eoigal committed:
fix space lint error

eoigal committed:
Merge branch 'update/reader-card-refresh' of github.com:Automattic/wp-calypso into update/reader-card-refresh

ebinnion committed:
Like Button: Add empty newline
